### PR TITLE
Add `nil` check for Burner

### PR DESCRIPTION
**Pull request by joshuahannan**
## Description

Add check to see if `AnyResource` is nil before burning it with the Burner

______

For contributor use:

- [x] Targeted PR against `v2-standard` branch
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: coverage.json
Comment (sisyphusSmiling) [inline coverage.json:1]: I think this file should be included in `.gitignore`. If so, np to include in separate PR

**Review by sisyphusSmiling — APPROVED** [review]
Looks good. Only note is unrelated - looks like the repo isn't configured for codecov which is an option now that we have Cadence tests. I can get that setup if you like, just lmk

File: coverage.json
Comment (joshuahannan) [reply coverage.json:1]: it is in there already, just hadn't removed it yet. I removed it

**Comment by joshuahannan** [comment]
I think this is fine for now. If we see that double optionals are causing problems in the future, then we can revisit

**Pull request merged**